app/models/webauthn_configuration.rb

```diff
@@ -3,6 +3,10 @@ class WebauthnConfiguration < ApplicationRecord
   validates :name, presence: true
   validates :credential_id, presence: true
   validates :credential_public_key, presence: true
+  validate :valid_transports
+
+  # https://w3c.github.io/webauthn/#enum-transport
+  VALID_TRANSPORTS = ['usb', 'nfc', 'ble', 'smart-card', 'hybrid', 'internal'].freeze
```

Comment:
personally I'd go for a `%w[]` array literal and put them on separate lines and alphabetize? Maybe even a `.to_set` since AFAICT order doesn't matter?

```suggestion
  VALID_TRANSPORTS = %w[
    ble
    hybrid
    internal
    nfc
    smart-card
    usb
  ].to_set.freeze
```

Reply:
The problem with converting to a set would be that `Array#difference` doesn't accept `Set` as an argument, so we'd have to cast it back to an array there in the validation. Maybe that's fine though, since it could be better to have the "source" be the type we prefer.

Reply:
but `-` works?

```ruby
irb(main):001:0> %w[a b c].to_set - %w[a b]
=> #<Set: {"c"}>
```

Reply:
oh but the opposite does not ;_;

```ruby
irb(main):002:0> %w[a b] - %w[a].to_set
(irb):2:in `-': no implicit conversion of Set into Array (TypeError)
```

Reply:
Seems okay enough to just cast it back to an array 🤷 Updated in 72b1837e4.